backend/src/routes/index.ts

```diff
@@ -0,0 +1,8 @@
+const routes = [
+  {
+    group: "user",
+    router: require("./user.route.ts").default,
```

Comment:
this makes the routing harder to follow. similar to my earlier comment. `route.group` assumes, a more strict rule of the array-based route definition, can't we work with an easier approach or use the previous setup.

a simple single export may also work, where everyone just define their routes here eg

```typescript
import express from "express";
import authRoute from "../user.route";
import userRoute from "../auth.route";
import settingsRoute from "../settings.route";
......

const router = express.Router();

router.use("/auth", authRoute);
router.use("/user", userRoute);
router.use("/settinfs", settingsRoute);
......

export default router;
```